main-option-2.py

```diff
@@ -0,0 +1,171 @@
+'''This Python script allows users to input a list of names and then randomly select a specified number of names from that list. 
+The program will prompt the user to enter names until they are ready to select a random name.'''
+
+import random
+import sys
+from subprocess import call
+
+num_input = 0
+name_input = None
+names = []
+
+#Re-run the app to play again
+def restartApp():
+    '''This function will restart the app if the user requests at the end of the previous game'''
+    call(["python", "main-option-2.py"])
+
+#Choose an amount of random outputs
+def addNum():
+    '''This function converts the string input into an int and uses it to select a random number of names from the list'''
+    num_input = input(f"Enter a number between 1 and {len(names)}: ")
+
+    validateNum(num_input)
+        
+    converted = float(num_input[0])
+
+    newN = int(converted)
+
+    while newN > len(names) or newN <= 0:
+
+        num_input = input(f"Number entered must be between 1 and {len(names)} inclusive: ")
+        
+        newN = int(num_input)
+
+    indices = []
+
+    for i in range(newN):
+
+        def newNum():
+
+            rNum = random.randint(1, len(names))
+
+            while rNum in indices:
+
+                rNum = random.randint(1, len(names))
+
+            else:
+
+                return rNum
+            
+        indices.append(newNum())
+
+    output = []
+
+    for j in indices:
+
+        output.append(names[j-1])
+
+    print(output)
+
+    again = input('Would you like to play again? Enter Y for yes or anything else to exit: ')
+
+    if again == 'Y' or again == 'y':
+        restartApp()
+    else:
+        sys.exit()  
+     
+    #return output
+
+#Add names to the list
+def enterName():
```

Comment:
This is the same logic as the code at the bottom of the file, there is no need to repeat the code.
This function could be called as entry into the programme.